### DISCUSSION TITLE
No such image: testcontainers/ryuk:0.3.4

### DISCUSSION THREAD

**Discussion by glendoci** [testcontainers/testcontainers-dotnet#659] [Q&A] [answered]
Getting the below issue when starting my containers was working fine last week, any ideas what could cause this? 

![image](https://user-images.githubusercontent.com/7105514/200286080-879bb68b-1044-4d48-aa3d-17128ab6c2f0.png)

**Comment by HofmeisterAn** [comment, marked as answer]
What happens if you pull the image via CLI?

**Reply by glendoci** [reply]
So pulling via CLI worked it stopped giving that error then after that I had the same issue with the Mongo image requiring 5.0.6 which I pulled manually again via the CLI and it worked. So it seems like it's failing to pull all required images that are missing...

Now I ran one of my tests and getting 
![image](https://user-images.githubusercontent.com/7105514/200299007-0820b178-7fd1-4d9e-ad90-69152b88436f.png)
Do I need to configure the a setting or something for testcontainers to properly talk to docker? Does test containers rely on the dockerflie? I did change some context stuff last week in the docker file. When I run the test I can see the containers being created so it does seem it's talking to the docker desktop.

My docker desktop is using the setting Use the WSL 2 based engine as well now.(could it have anything to do with it)

No clue what's happening or what changed.

**Reply by HofmeisterAn** [reply]
> Do I need to configure the a setting or something for testcontainers to properly talk to docker? Does test containers rely on the dockerflie?

I don't know what you're talking about, but Testcontainers tries to detect the Docker endpoint. To customise it, take a look at our [docs](https://dotnet.testcontainers.org/custom_configuration/).

> I did change some context stuff last week in the docker file.

Can you share your Dockerfile? Are you connected to a private registry?

**Reply by glendoci** [reply]
FROM mcr.microsoft.com/dotnet/aspnet:6.0 AS base
WORKDIR /app

FROM mcr.microsoft.com/dotnet/sdk:6.0 AS build
WORKDIR /src
COPY ["./jd.prism.privateapigateway/src/Jd.Prism.PrivateApiGateway.WebApi/Jd.Prism.PrivateApiGateway.WebApi.csproj", "Jd.Prism.PrivateApiGateway.WebApi/"]
COPY ["./jd.prism.privateapigateway/src/Jd.Prism.PrivateApiGateway.Mongo/Jd.Prism.PrivateApiGateway.Mongo.csproj", "Jd.Prism.PrivateApiGateway.Mongo/"]
COPY ["./jd.prism.privateapigateway/src/Jd.Prism.PrivateApiGateway.Domain/Jd.Prism.PrivateApiGateway.Domain.csproj", "Jd.Prism.PrivateApiGateway.Domain/"]
COPY ["./jd.prism.common.netcore/src/Jd.Prism.Common.NetCore/Jd.Prism.Common.NetCore.csproj", "Jd.Prism.Common.NetCore/"]
COPY ["./jd.prism.common.core/src/Jd.Prism.Common.Core/Jd.Prism.Common.Core.csproj", "Jd.Prism.Common.Core/"]
COPY . ./

RUN dotnet restore "./jd.prism.privateapigateway/src/Jd.Prism.PrivateApiGateway.WebApi/Jd.Prism.PrivateApiGateway.WebApi.csproj"
RUN dotnet build "./jd.prism.privateapigateway/src/Jd.Prism.PrivateApiGateway.WebApi/Jd.Prism.PrivateApiGateway.WebApi.csproj" -c Release -o /app/build

FROM build AS publish
RUN dotnet publish "./jd.prism.privateapigateway/src/Jd.Prism.PrivateApiGateway.WebApi/Jd.Prism.PrivateApiGateway.WebApi.csproj" -c Release -o /app/publish

FROM base AS final
WORKDIR /app

RUN apt-get update && apt-get upgrade -y && apt-get install -y
RUN apt-get install nano -y
RUN apt-get install curl -y
RUN apt-get install iputils-ping -y
RUN apt-get install telnet -y
RUN apt-get install net-tools -y
RUN apt-get install exiftool -y

COPY --from=publish /app/publish .
ENTRYPOINT ["dotnet", "Jd.Prism.PrivateApiGateway.WebApi.dll"]

I am using docker compose to run multiple services with Rider, that's the docker file of the service I am trying to run the tests on.

**Reply by HofmeisterAn** [reply]
I meant the changes you mentioned here:

> I did change some context stuff last week in the docker file.

**Reply by glendoci** [reply]
I doubt they will have anything to do with it it's just what rider will use to run my services in docker and how they are copied over.

**Reply by HofmeisterAn** [reply]
Did you enable the beta feature to use containerd for pulling and storing images?

**Reply by glendoci** [reply]
So the pulling is still not working but I managed to find whats causing this 
![image](https://user-images.githubusercontent.com/7105514/200314858-89e9b83e-89fe-4e53-8786-1210979f0699.png)
and that is the Use the WSL 2 based engine setting on which I need on to debug my microservices on Rider 
![image](https://user-images.githubusercontent.com/7105514/200314939-9b0f17b2-23c8-4112-bef1-f1202b497c89.png)

**Reply by glendoci** [reply]
Could that setting have anything to do with the rest too?

**Reply by HofmeisterAn** [reply]
> So the pulling is still not working but I managed to find whats causing this

What's the issue?

> and that is the Use the WSL 2 based engine setting on which I need on to debug my microservices on Rider

This setting is fine.

> Could that setting have anything to do with the rest too?

No, WSL is supported and works fine. I use it too. What did you change within the last week? I mean you said it was working?

**Reply by glendoci** [reply]
Tbh nothing else but the context in docker compose   
build:
     context: .
to     
build:
      context: ./

**Reply by HofmeisterAn** [reply]
Sorry, I don't understand what you are trying to achieve, I am not familiar with this setup. There is no Docker Compose support in Testcontainers for .NET. It's using the daemon socket to communicate with Docker. What does `TestcontainersSettings.OS.DockerEndpointAuthConfig.Endpoint` resolves to? That's the endpoint Testcontainers for .NET use to communicate with Docker.

**Reply by glendoci** [reply]
Started working again out of thin air... no changes
